planner/core/find_best_task.go

```diff
@@ -1330,11 +1402,42 @@ func (ds *DataSource) buildIndexMergeTableScan(_ *property.PhysicalProperty, tab
 			}
 			sel := PhysicalSelection{Conditions: pushedFilters}.Init(ts.ctx, ts.stats.ScaleByExpectCnt(selectivity*totalRowCount), ts.blockOffset)
 			sel.SetChildren(ts)
-			return sel, remainingFilters, nil
+			return sel, remainingFilters, false, nil
+		}
+		return ts, remainingFilters, false, nil
+	}
+	// If there's filters on table side, the limit cannot be pushed to index side.
+	// So we just need to check whether the prop is matched after checking the table filters.
```

Comment:
I think this comment is quite confusing, the checking is only related to whether `keep order` is needed, it's irrelevant to the `Limit` pushdown.